### DISCUSSION TITLE
About the example for 'Responsive Everythins'.

### DISCUSSION THREAD

**Discussion by yuzhou1990** [tailwindlabs/tailwindcss#13491] [Help]
![image](https://github.com/tailwindlabs/tailwindcss/assets/38753654/f3a5b830-a5b4-41b3-af30-56abcc402e82)

Why is the content on the left vertically centered in the grid?

Is it caused by some properties of the grid?

Can someone help me?

**Comment by alex-krasikau** [comment]
Hey @seventh1990!

> There's a bug in how the browser displays row indexes on this screenshot. It should show `1, 2, 3, 4...` instead of `1, 1, 2, 3...`.

As you can see in the markup, each child is assigned specific positions in the grid using `col-start-*`, `col-end-*`, `row-start-*`, and `row-end-*` utilities. In the final desktop layout, cells with row indexes 1 to 2 (1 to 1 in the screenshot) and 6 to 7 (5 to 6 in the image) are empty.

When the browser encounters empty cells in the grid without a `grid-template-rows` declaration, it sets `height: auto` for these cells. And if there is extra space left, it evenly distributes available space across empty cells, resulting in a centered layout in this scenario.

**Reply by yuzhou1990** [reply]
Thank you very much for your reply.

The example above is from the official website.

Here's my simplified sample code:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
    <script src="https://cdn.tailwindcss.com"></script>
</head>
<body>
    <main>
        <div class="mx-auto grid max-w-5xl gap-x-20 grid-cols-2">
          <div class="col-start-1 row-start-1">
            <p>1</p>
          </div>
          <div class="grid col-end-3 grid-cols-4 gap-6 col-start-2 row-span-6 row-end-6">
            <img src="/beach-house.jpg" alt="" class="w-full h-60 object-cover rounded-lg lg:col-span-full" loading="lazy">
            <img src="/beach-house-interior-1.jpg" alt="" class="w-full object-cover rounded-lg lg:row-start-2 lg:col-span-2 lg:h-32" loading="lazy">
            <img src="/beach-house-interior-2.jpg" alt="" class="w-full object-cover rounded-lg lg:row-start-2 lg:col-span-2 lg:h-32" loading="lazy">
          </div>
          <dl class="row-start-2 lg:row-start-2">
            1
          </dl>
          <div class="col-start-1 row-start-3 row-span-2 lg:col-start-1 lg:row-start-3 lg:row-end-4">
            <button>1</button>
          </div>
          <p class="col-start-1 lg:row-start-4 lg:col-span-1">
            1
          </p>
        </div>
      </main>
</body>
</html>
```

After many attempts, I finally discovered the reason.

Moreover, I don't think there's a display bug with row indexes. On the contrary, it's precisely by utilizing defined row indexes that the above effect is achieved.

The key to this piece of code lies in the `row-end-6` style. It changes the grid line indexes by defining the end row indexes of the content on the right as **6**, thus achieving the final effect.

The CSS code that actually takes effect：

```css
grid-row-start: span 6;
grid-row-end: 6;
```

You can try changing `row-end-6` to `row-end-10` or some other number and see the effect.

Although I found the reason, I haven't found an explanation for this feature.

**Reply by alex-krasikau** [reply]
You're correct that `lg:row-end-6 lg:row-span-6` causes the issue by shifting the image block to row `-1`. Using just `lg:row-span-6` would position the elements correctly. However, the current setup needs adjusting all `row-start-*` declarations and increasing the index by 1.

Here's a simplified example I created: [Play Tailwind - Grid](https://play.tailwindcss.com/eWC4iaPApY). The rows have the correct indexes marked. Does this help?

<img width="636" alt="CleanShot 2024-04-10 at 10 52 47@2x" src="https://github.com/tailwindlabs/tailwindcss/assets/97882909/0864ad80-d348-4783-8a38-12683e12a97b">

**Reply by yuzhou1990** [reply]
Thanks.

Your code is good and conforms to the normal logic of grids.

I initiated this discussion only because I have some doubts about the implementation on the official website.
